app/uk/gov/hmrc/preferencesadminfrontend/controllers/MessageBrakeController.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.preferencesadminfrontend.controllers
+
+import javax.inject.Inject
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, Request, Result}
+import uk.gov.hmrc.play.frontend.controller.FrontendController
+import uk.gov.hmrc.preferencesadminfrontend.config.AppConfig
+import uk.gov.hmrc.preferencesadminfrontend.connectors.MessageConnector
+import uk.gov.hmrc.preferencesadminfrontend.model.GmcBatch
+import uk.gov.hmrc.preferencesadminfrontend.services.MessageService
+import uk.gov.hmrc.preferencesadminfrontend.views.html.{error_template, message_brake_admin}
+
+class MessageBrakeController @Inject()(messageConnector: MessageConnector, messageService: MessageService)
+                                      (implicit appConfig: AppConfig, val messagesApi: MessagesApi) extends FrontendController with I18nSupport {
+
+  def showAdminPage: Action[AnyContent] = AuthorisedAction.async {
+    implicit request =>
+      implicit user =>
```

Comment:
It looks like "implicit user " is redundant in all methods. It is not used anywhere...

Reply:
it's used in AuthorisedAction